pkg/controller/nodeipam/ipam/multinetwork_cloud_cidr_allocator.go

```diff
@@ -0,0 +1,62 @@
+package ipam
+
+import (
+	"context"
+	"fmt"
+
+	compute "google.golang.org/api/compute/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	networkv1 "k8s.io/cloud-provider-gcp/crd/apis/network/v1"
+)
+
+func (ca *cloudCIDRAllocator) PerformMultiNetworkCIDRAllocation(node *v1.Node, interfaces []*compute.NetworkInterface) (defaultNwCIDRs []string, northInterfaces networkv1.NorthInterfacesAnnotation, additionalNodeNetworks networkv1.MultiNetworkAnnotation, err error) {
+	k8sNetworksList, err := ca.networkClient.NetworkingV1().Networks().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return nil, nil, nil, fmt.Errorf("error fetching networks: %v", err)
+	}
+	k8sNetworks := make([]networkv1.Network, 0)
+	// ignore networks that are under deletion.
+	for _, network := range k8sNetworksList.Items {
+		if network.ObjectMeta.DeletionTimestamp.IsZero() {
+			k8sNetworks = append(k8sNetworks, network)
+		}
+	}
+	gkeNwParamsClient := ca.networkClient.NetworkingV1().GKENetworkParamses("")
+	// Fetch the GKENetworkParams for every k8s-network object.
+	// Match the fetched GKENetworkParams object with the interfaces on the node
+	// to build the per-network north-interface and node-network annotations useful for IPAM.
+	for _, network := range k8sNetworks {
+		gnp, err := gkeNwParamsClient.Get(context.TODO(), network.Spec.ParametersRef.Name, metav1.GetOptions{})
+		if err != nil {
+			return nil, nil, nil, err
+		}
+		for _, inf := range interfaces {
+			if inf.Network != gnp.Spec.VPC || inf.Subnetwork != gnp.Spec.VPCSubnet {
+				continue
```

Comment:
add some logging with verbosity for debugging?

Reply:
Done.